src/main/java/org/jabref/gui/externalfiles/FileNameParser.java

```diff
@@ -0,0 +1,21 @@
+package org.jabref.gui.externalfiles;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class FileNameParser {
+
+    private static final Pattern FIELD_PATTERN = Pattern.compile("\\[([^\\]]+)\\]");
```

Comment:
I'm pretty sure this is already implemented. Just to clarify, it is possible to change the linked file name to follow a pattern by right-clicking on the linked file and choosing 'rename file to a defined pattern'.

Reply:
Changed the code to use BracketedPattern instead and deleted this class!!